winsup/cygwin/include/cygwin/exit_process.h

```diff
@@ -71,11 +74,11 @@ get_address_from_cygwin_console_helper(int bitness, wchar_t *function_name)
 
   STARTUPINFOW si = {};
   PROCESS_INFORMATION pi;
-  size_t len = wcslen (wbuf) + wcslen (function_name) + 1;
+  size_t len = wcslen (wbuf) + 1 + wcslen (function_name) + 1 + (exit_code == 0 ? 1 : (int)(log10 (exit_code) + 1)) + 1 + (pid == 0 ? 1 : (int)(log10 (pid) + 1));
```

Comment:
Oy! That's a lot of CPU spent on calculating the size we need to allocate to fix the exit code.

Since exit codes are limited to three digits (7 bits, to be precise, or 8 bits if you accommodate for signals), and process IDs are [limited to `DWORD`](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getprocessid) (AKA 32-bit `unsigned long`) all we need is to change that `+ 1` to `+ 3 /* exit code */ + 1 /* space */ + 10 /* process ID, i.e. DWORD */ + 1 /* NUL */;`

Reply:
Done https://github.com/msys2/msys2-runtime/pull/31/commits/cc10c797854fa2c4086b493eba20856738a8934c